dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -1236,95 +1245,187 @@ def read_05evt_c_file(self):
     # @param self : object pointer
     # @return
     def read_struct_xml_file(self):
-        doc = minidom.parse(self.struct_xml_file)
-        root = doc.documentElement
-        # read file structures
-        elements = root.getElementsByTagName('elements')
+        xml_tree = lxml.etree.parse(self.struct_xml_file)
+        root_element = xml_tree.getroot()
+        if root_element.tag != "model":
+            raise ModelElementNotFound(root_element.tag)
 
-        # list of structures/terminals encountered during reading
         list_struct = []
-        self.list_elements=[]
-        if(len(elements) > 0 ):
-            for node_element in elements:
-                # reading structures
-                nodes_struct = node_element.getElementsByTagName('struct')
-                for node in nodes_struct:
-                    struct_name = node.getElementsByTagName('name')[0].firstChild.nodeValue
-                    node_terminal = node.getElementsByTagName('terminal')[0]
-                    name_terminal = node_terminal.getElementsByTagName('name')[0].firstChild.nodeValue
-                    connector_node =  node_terminal.getElementsByTagName('connector')[0]
-                    type_connector = str(connector_node.getAttribute('type'))
-                    compo_name=struct_name + "."+name_terminal
-
-                    # we store all this in an object
-                    structure = Element(True,struct_name,len(self.list_elements))
-                    if struct_name not in list_struct:
-                        list_struct.append(struct_name)
-                        self.list_elements.append(structure)
-                        self.map_struct_name_struct_obj[struct_name] = structure
+        self.list_elements = []
+
+        for model_element in root_element:
+            if model_element.tag == "name":
+                pass  # nothing to do
+            elif model_element.tag == "elements":
```

Comment:
Here structs are passed before terminals as before. Not sure it's necessary tho.